src/Analyzer.JsonRuleEngine/Operators/RegexOperator.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Templates.Analyzer.RuleEngines.JsonEngine.Operators
+{
+    internal class RegexOperator : LeafExpressionOperator
+    {
+        /// <inheritdoc/>
+        public override string Name => "Regex";
+
+        /// <summary>
+        /// Gets the regex pattern this operator will compare against.
+        /// </summary>
+        public string RegexPattern { get; private set; }
+
+        /// <summary>
+        /// Creates a RegexOperator.
+        /// </summary>
+        /// <param name="regexPattern">The regex pattern specified in the JSON rule.</param>
+        /// <param name="isNegative">Whether the result of EvaluateExpression() should be negated or not.</param>
+        public RegexOperator(string regexPattern, bool isNegative)
```

Comment:
I think for regex it doesn't make sense to have an `isNegative` parameter.  This can be removed, and `this.IsNegative` can just always be `false`.

Reply:
I assumed we were going to use isNegative for when the not operator is implemented.

Reply:
I think `not` has to be its own `Expression` that 'not's the expression inside it.  The language supports doing something like `"not": { "allOf": [...] }` too, which wouldn't have any way to 'negate' the `allOf` other than having `not` produce its own Evaluation.

Reply:
I guess it also makes sense since technically doing not { notEquals } would make the isNegative property more complicated.